pkg/kubelet/network/kubenet/kubenet_linux.go

```diff
@@ -281,6 +288,24 @@ func (plugin *kubenetNetworkPlugin) SetUpPod(namespace string, name string, id k
 		return err
 	}
 
+	// Put the container bridge into promiscuous mode to force it to accept hairpin packets.
+	// TODO: Remove this once the kernel bug (#20096) is fixed.
+	if plugin.wantPromisc && !plugin.havePromisc {
```

Comment:
I'll just change it like you suggest, actually.  Once the CNI PR lands we can remove this (and a few other things) anyway.